src/Connection.php

```diff
@@ -327,13 +345,15 @@ protected function getDefaultPostProcessor()
     /** @inheritdoc */
     protected function getDefaultQueryGrammar()
     {
-        return new Query\Grammar();
+        // Argument added in Laravel 12
+        return new Query\Grammar($this);
```

Comment:
I assume this relies on the relative namespace of the file. Is that preferable to explicitly using these classes? Perhaps, since doing so would require aliases.

Reply:
In find it more explicit than having to alias one of `Query\Grammar` and `Schema\Grammar`.